lib/Analysis/Performance.swift

```diff
@@ -0,0 +1,46 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+/// IdentifierFactory creates unique identifiers for given elements.
+/// It provides an API to map between identifiers and elements and can be used
+/// for fast lookups.
+public final class IdentifierFactory<T: Hashable> {
```

Comment:
Can we just call this file `IdentifierFactory`?

Reply:
Yeah, definitely. I had some other stuff in here as well which got removed during iterations. Thanks